tedana/interfaces/tedana.py

```diff
@@ -399,28 +498,25 @@ def fitmodels_direct(catd, mmix, mask, t2s, t2sG, tes, combmode, head,
     totvar = (tsoc_B**2).sum()
     totvar_norm = (WTS**2).sum()
 
-    # Compute Betas and means over TEs for TE-dependence analysis
-    n_echoes = len(tes)
-    betas = cat2echos(get_coeffs(uncat2echos(catd, n_echoes),
-                                 np.tile(mask, (1, 1, n_echoes)),
-                                 mmix), n_echoes)
-    nx, ny, nz, _, n_components = betas.shape
+    # compute Betas and means over TEs for TE-dependence analysis
+    betas = get_coeffs(catd, np.repeat(mask[:, np.newaxis], len(tes), axis=1), mmix)
+    n_samp, n_echos, n_components = betas.shape
     n_voxels = mask.sum()
     n_data_voxels = (t2s != 0).sum()
-    mu = catd.mean(axis=-1)
-    tes = np.reshape(tes, (n_echoes, 1))
-    fmin, fmid, fmax = getfbounds(n_echoes)
+    mu = catd.mean(axis=-1, dtype=float)  # BUG: THIS IS THE BAD PLACE
```

Comment:
Yes, this can definitely be removed since there is a trace of the rationale behind changing it. It might be worth following numpy development and reverting this change eventually, but I think that will be further down the line.

Reply:
Yes, this can definitely be removed since there is a trace of the rationale behind changing it. It might be worth following numpy development and reverting this change eventually, but I think that will be further down the line.